homeassistant/helpers/trigger_template_entity.py

```diff
@@ -176,35 +254,94 @@ def restore_attributes(self, last_state: State) -> None:
                 extra_state_attributes[attr] = last_state.attributes[attr]
             self._rendered[CONF_ATTRIBUTES] = extra_state_attributes
 
-    def _render_templates(self, variables: dict[str, Any]) -> None:
-        """Render templates."""
+    def _template_variables(self, run_variables: dict[str, Any] | None = None) -> dict:
+        """Render template variables."""
+        return {
+            "this": TemplateStateFromEntityId(self.hass, self.entity_id),
+            **(run_variables or {}),
+        }
+
+    def _render_single_template(
+        self,
+        key: str,
+        variables: dict[str, Any],
+        strict: bool = False,
+    ) -> Any:
+        """Render a single template."""
         try:
-            rendered = dict(self._static_rendered)
+            if key in self._to_render_complex:
+                return render_complex(self._config[key], variables)
 
-            for key in self._to_render_simple:
-                rendered[key] = self._config[key].async_render(
-                    variables,
-                    parse_result=key in self._parse_result,
-                )
+            return self._config[key].async_render(
+                variables, parse_result=key in self._parse_result, strict=strict
+            )
+        except TemplateError as err:
+            log_triggered_template_error(self.entity_id, err, key=key)
 
-            for key in self._to_render_complex:
-                rendered[key] = render_complex(
-                    self._config[key],
-                    variables,
-                )
+        return _SENTINEL
+
+    def _render_availability_template(self, variables: dict[str, Any]) -> bool:
+        """Render availability template."""
+        if not self._availability_template:
+            return True
 
-            if CONF_ATTRIBUTES in self._config:
-                rendered[CONF_ATTRIBUTES] = render_complex(
-                    self._config[CONF_ATTRIBUTES],
-                    variables,
+        try:
+            if (
+                available := self._availability_template.async_render(
+                    variables, parse_result=True, strict=True
                 )
+            ) is False:
+                self._rendered = dict(self._static_rendered)
+
+            self._available = available
 
-            self._rendered = rendered
         except TemplateError as err:
-            logging.getLogger(f"{__package__}.{self.entity_id.split('.')[0]}").error(
-                "Error rendering %s template for %s: %s", key, self.entity_id, err
-            )
-            self._rendered = self._static_rendered
+            self._available = True
```

Comment:
I think we should motivate in a comment why the entity is considered available in case rendering the template fails